core/unit_test/CMakeLists.txt

```diff
@@ -338,6 +339,35 @@ KOKKOS_ADD_ADVANCED_TEST( UnitTest_PushFinalizeHook_terminate
     ALWAYS_FAIL_ON_ZERO_RETURN
 )
 
+if(Kokkos_ENABLE_PROFILING)
+  add_library(
+     printer-tool SHARED
+     tools/printing-tool.cpp
+  )
+
+  KOKKOS_ADD_TEST_EXECUTABLE(
+    ProfilingAllCalls
+    tools/TestAllCalls.cpp
+  )
+
+  KOKKOS_ADD_TEST( NAME ProfilingTestParallelFor
+    EXE  ProfilingAllCalls
+    TOOL ${CMAKE_CURRENT_BINARY_DIR}/libprinter-tool.so
+    PASS_REGULAR_EXPRESSION ".*kokkosp_begin_parallel_for:parallel_for:[^:]*:0.*end_parallel_for:0"
+  )
+  KOKKOS_ADD_TEST( NAME ProfilingTestParallelReduce
+    EXE  ProfilingAllCalls
+    TOOL ${CMAKE_CURRENT_BINARY_DIR}/libprinter-tool.so
+    PASS_REGULAR_EXPRESSION ".*kokkosp_begin_parallel_reduce:parallel_reduce:[^:]*:1.*end_parallel_reduce:1"
+  )
+  KOKKOS_ADD_TEST( NAME ProfilingTestParallelScan
+    EXE  ProfilingAllCalls
+    TOOL ${CMAKE_CURRENT_BINARY_DIR}/libprinter-tool.so
+    PASS_REGULAR_EXPRESSION ".*kokkosp_begin_parallel_scan:parallel_scan:[^:]*:2.*end_parallel_scan:2"
```

Comment:
What we are doing in other projects is to save the output and compare it using an external tool like `diff` or `numdiff`. Of course, this requires that we don't record runtime specific data.

Reply:
Could you link me to a CMake example of this? I like that approach where it's applicable, and I think it's applicable for many of the tests I plan to write

Reply:
Basically, if we're building up unit testing for the subsystem from the ground up, I'd like to do it right, and I like what you're suggesting there. I don't want to build up a big testing system that's wrong and then have to change it

Reply:
Basically, it's https://github.com/dealii/dealii/blob/dd5c780f138ffb3260c16b7dc8bc1b61c6a7433b/cmake/macros/macro_deal_ii_add_test.cmake#L283-L340
using https://github.com/dealii/dealii/blob/master/cmake/scripts/run_test.cmake and https://github.com/dealii/dealii/blob/master/cmake/scripts/run_test.sh
It might not be straightforward to adapt, though.

Reply:
I'll bring it up to the group. Since `Tools` isn't the core of what we do, I don't want to overengineer this and create too big a maintenance burden, but if the team thinks it's worthwhile then I will go ahead with it. Thanks for the info

Reply:
My compromise answer here was just to make a big regex that matches the whole string. As this is now just testing the library load portion, I think it's a good way to work